File: Python/50_PowXN.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0: return 1
        if n == 1: return x
        if n == -1: return 1/x
        halfies = self.myPow(x, n/2) if n % 2 == 0 else self.myPow(x, (n-1)/2)
        if n % 2 == 0: return halfies * halfies
        else: return halfies * halfies * x
        

File: C#/50_PowXN.cs
public class Solution {
    public double MyPow(double x, int n) {
        if (n == 0) return 1; 
        if (n == 1) return x;
        if (n == -1) return 1/x;    
        double halfie = n % 2 == 0? MyPow(x, n/2) : MyPow(x, (n-1)/2);
        if (n % 2 == 0) return halfie * halfie;
        else return halfie * halfie * MyPow(x,1);
    }
}
